tests/streaming/test_dataset.py

```diff
@@ -819,7 +820,7 @@ def test_dataset_resume_on_future_chunks(tmpdir, monkeypatch):
 
     optimize(
         fn=_simple_preprocess,
-        inputs=list(range(8)),
+        inputs=list(range(5)),
```

Comment:
Why was this number changed @deependujha? 
This example was constructed in a specific way.

Reply:
@deependujha Unfortunately, we found another core issue and this seemed to be a legit test failure.